app/Models/Asset.php

```diff
@@ -637,6 +637,25 @@ public function checkins()
             ->withTrashed();
     }
 
+    public function lastCheckin()
+    {
+        return $this->hasOne(Actionlog::class, 'item_id')
+            ->ofMany(
+                ['action_date' => 'max'],
+                function ($query) {
+                    $query->where('item_type', '=', self::class)
+                        ->where('action_type', '=', 'checkin from');
+                })
+            ->withTrashed();
+    }
+
+    public function scopeLastCheckinBetween($query, $start, $end)
```

Comment:
Are we only using this here? We could probably just scope this in the API query itself, but that's not a hill I'll die on :)

Reply:
No worries. I inlined it 👍🏾